File: src/test/java/com/claudio_personagens/bookingcharactersapi/PostgresqlContainer.java
package com.claudio_personagens.bookingcharactersapi;

import org.testcontainers.containers.PostgreSQLContainer;

public class PostgresqlContainer extends PostgreSQLContainer<PostgresqlContainer> {

    private static final String IMAGE_VERSION = "postgres:10.6";
    private static PostgresqlContainer container;

    private PostgresqlContainer(){
        super(IMAGE_VERSION);
    }

    public static PostgresqlContainer getInstance() {
        if (container == null) {
            container = new PostgresqlContainer();
        }
        return container;
    }

    @Override
    public void start() {
        super.start();
        System.setProperty("spring.datasource.url", container.getJdbcUrl());
        System.setProperty("spring.datasource.password", container.getUsername());
        System.setProperty("spring.datasource.username", container.getPassword());
    }
}

File: src/test/java/com/claudio_personagens/bookingcharactersapi/PostgresqlContainer.kt
package com.claudio_personagens.bookingcharactersapi

import org.testcontainers.containers.PostgreSQLContainer

class PostgresqlContainer private constructor() : PostgreSQLContainer<PostgresqlContainer>(IMAGE_VERSION) {

    companion object {
        private const val IMAGE_VERSION = "postgres:10.6"
        private var container: PostgresqlContainer = PostgresqlContainer()
        val instance: PostgresqlContainer
            get() {
                return container
            }
    }

    override fun start() {
        super.start()
        System.setProperty("spring.datasource.url", container.jdbcUrl)
        System.setProperty("spring.datasource.password", container.username)
        System.setProperty("spring.datasource.username", container.password)
    }
}